rest/handler.go

```diff
@@ -473,7 +479,7 @@ func (h *handler) validateAndWriteHeaders(method handlerMethod, accessPermission
 			}
 			// The above readBody() will end up clearing the body which the later handler will require. Re-populate this
 			// for the later handler.
-			h.requestBody = io.NopCloser(bytes.NewReader(body))
+			h.requestBody.reader = io.NopCloser(bytes.NewReader(body))
```

Comment:
I suspect this will cause double counting of the request body, as we're basically repopulating the counter's reader to be consumed again. Could do with writing a test (likely an integration test for RBAC) that hits this codepath to verify.

Seems like we need to replace the `CountedRequestReader` with a non-counting `io.Reader` (which has implications for the type of `h.requestBody`.